text/0009-golang.md

```diff
@@ -0,0 +1,68 @@
+ 
+The simplicity of Go stems from the fact that, unlike Java, it is not a multi-purpose tool. Go's main uses cases are web services and CLIs (command-line interfaces) which makes it a perfect fit for writing FOLIO backend modules. Being simpler than Java, the language is also easier to pick up by full-stack developers coming from scripting and web languages background like Node.js or Python. This can be a big advantage for small teams of few developers as the entire team can, effectively, be kept full-stack. With large Java codebases it’s not practical and developers in the team usually specialize in front-end or back-end programming. 
+
+On the operational front, Go's characteristics make it a better fit than Java for building systems comprised of many small microservices. Go is compiled ahead-of-time (AOT) into a static binary which is a perfect fit for Docker containers. AOT compilation means that the resulting application has a visibly smaller memory footprint and faster startup times as compared to Java/vert.x, and orders of magnitude better as compared to Java/Spring based modules. Java provides AOT compilation capability through the GraalVM project, but porting existing Java projects to GraalVM involves substantial complexity and is not possible for certain frameworks like older Spring (including versions used by FOLIO project) and Grails due to the dynamic language features they require. 
+
+From a sysops perspective, Go containers require less operational tuning compared to the JVM containers as they do not contain a stand-alone virtual machine that requires parametrization.
+
+The Go language is widely used across the industry and open-source communities, especially in the areas related to web and microservices. Popular examples of large open-source projects created with Go include the [Kubernetes](https://github.com/kubernetes/kubernetes) container orchestration platform and the [Docker](https://github.com/docker) container runtime.
+
+The Go project maintains a [list](https://go.dev/solutions/) of case-studies from commercial companies relying on Go in their products and general use-cases for the language.
+
+According to [GitHub statistics](https://madnight.github.io/githut/#/pull_requests/2023/3), Go is the third fastest growing language (behind Python and Java) as measured by the number of new Pull Requests.
+
+## Risks and Drawbacks
+
+Introducing Go as one of the supported backend technologies involves the same general concerns as introducing any new technology into FOLIO. The biggest potential risk is the increased development and maintenance complexity as some new FOLIO backend modules will be created with Go which is not immediately well known to existing FOLIO backend developers that have primarily Java background. As explained in the previous section, however, this risk is highly minimized because of the design of Go which has many points of similarity to Java.
```

Comment:
> As explained in the previous section, however, this risk is highly minimized because of the design of Go which has many points of similarity to Java.

Personal opinion (from someone who likes Go): "highly minimized" feels like a bit of a stretch. Go is a great language, but there are quite a few common idioms that could make some code tricky to follow for someone new to the language.
Here's a short list of things that stand out as things that are pretty radically different from Java (just from a quick perusal of Effective Go), where a Java dev might need to go track down docs just to understand code they are reading:
- `=` vs `:=`
- error handling and `panic`/`recover`
- every loop is a `for`-loop
- `range`
- no fallthrough in `switch` blocks
- non-C-like pointers
- the variable declaration syntax
- functions with naked return statements even though they return values
- deferred functions
- the syntax around type parameters when declaring stuff like arrays and maps
- having to use `make` to allocate some objects, but not others
- slices
- `_` (oh, how I wish Java had `_`...)
- semantics around capitalization
- Go's enums are kind of weird compared to Java's
- `init()`
- `interface {}`
- the standard lib's interfaces can take some getting used to (e.g., `io.Reader` and `io.Writer`)
- goroutines ❤️ 
- channels ❤️ 

(note: this could also be considered a list of features that make Go pretty awesome 🤷)

Reply:
Well, this is highly subjective but from my personal experience, I think it mostly holds.  The syntax differences you list are of course there but in most cases it's "cosmetics" and not paradigm shifting changes. I'd say the biggest difference is the concurrency model (goroutines) and more advanced topics like generics. Though concurrency is already sufficiently different between traditional thread-base Spring and async/future-based vert.x to make this area more generally complex. Also, pointers and stack vs heap allocation is something that may be unfamiliar to Java programmers, especially with no background in C/C++.

I'll rephrase this to acknowledge some major difference but without getting too much into details.